include/sound/sof.h

```diff
@@ -17,6 +17,28 @@
 
 struct snd_sof_dsp_ops;
 
+/**
+ * enum snd_sof_fw_state - DSP firmware state definitions
+ * @SOF_FW_BOOT_NOT_STARTED:	firmware boot is not yet started
+ * @SOF_FW_BOOT_PREPARE:	preparing for boot (firmware loading for exaqmple)
+ * @SOF_FW_BOOT_IN_PROGRESS:	firmware boot is in progress
+ * @SOF_FW_BOOT_FAILED:		firmware boot failed
```

Comment:
i know it makes difference functionally, but can we move BOOT_FAILED to after READY_OK? so that we can kind of get a sense of progression

Reply:
The boot failed can happen only before `BOOT_READY_FAILED` or `BOOT_READY_OK`, it is the state which indicates that the fw_ready message has not arrived.

Reply:
so are we missing a state between READY_OK and BOOT_COMPLETE. The only way this state would be reached is if post_fun_run failed.

Reply:
sounds good to me.

Reply:
@ranj063, great!
One more thing: should I push a new patch to this PR for:
"move to `SOF_FW_BOOT_FAILED` if `snd_sof_dsp_post_fw_run() fail."
Or can I leave it for the follow-up state handling cleanup?

Reply:
lets handle that in the follow-up too and make it complete.